Realm/Realm/DatabaseTypes/RealmObjectBase.cs

```diff
@@ -501,5 +506,99 @@ public Metadata(TableKey tableKey, IRealmObjectHelper helper, IDictionary<string
                 Schema = schema;
             }
         }
+
+        /// <summary>
+        /// A class that exposes a set of API to access the data in a managed RealmObject dynamically.
+        /// </summary>
+        public struct Dynamic
```

Comment:
@papafe @fealebenpae @LaPeste @DominicFrei - what do you folks think about this departure from the proposed design? I moved all the string-based API in a new object so they are "namespaced" together:

```csharp
RealmObject foo = ...;

foo.DynamicApi.Get<int>("MyIntProp");
foo.DynamicApi.Set("MyStringProp", "abc");
```

The main motivation there is that I didn't want those to pollute the user models as these API would have been suggested along with the original user properties even though in 99% of the cases, the user wouldn't need them. The alternatives I considered are:
* Keep them exposed directly on `RealmObjectBase` but annotate with `[EditorBrowsable(Never)]`. The reason why I didn't like this approach was that we would make them quite a bit less discoverable and people would need to consult the docs for every single thing.
* Have RealmObjectBase implement some special interface explicitly and ask people to cast to that. I didn't like this approach a lot because we couldn't declare a method as returning the union of RealmObjectBase, meaning that it would always be one or the other and users would need to cast back and forth depending on whether they need the RealmObject API or the dynamic ones.
* Have just `object Get(string propName)` and `Set(string propName, object)` that would handle lists, sets, singular values and so on. That wasn't great either because it requires quite a bit of casting on the user side and quite a bit of guessing - do I cast to `IList`/`List`/`RealmList` and so on.
* Have some RealmObject inheritor that exposes these methods directly and have the dynamic methods return that. I decided against that because in some cases, the dynamic API are actually returning the user-defined models and that would be impossible if we also needed to return the special type.

Reply:
Looking into the options the reasons against them I think the chosen solution seems to be the best way too.
Also, big fan of namespaces as they often improve readability. 👍

Reply:
I also agree that this seems to be the best option, especially considering it's something that is "rarely" used.

Reply:
I fully agree with the exposed reasoning. It's nice touch given the rarity of usage.